frontend/src/pages/CompareV2.tsx

```diff
@@ -292,9 +313,74 @@ function getDropdownItems(filteredRunArtifacts: RunArtifact[]) {
   return dropdownItems;
 }
 
+function getArtifactFromSelectedItem(
+  filteredRunArtifacts: RunArtifact[],
+  selectedItem: SelectedItem,
+): Artifact | undefined {
+  const filteredRunArtifact = filteredRunArtifacts.find(
+    runArtifact => runArtifact.run.run?.name === selectedItem.itemName,
+  );
+
+  const executionArtifact = filteredRunArtifact?.executionArtifacts.find(executionArtifact => {
+    const executionText: string =
+      getExecutionName(executionArtifact.execution) ||
+      executionArtifact.execution.getId().toString();
```

Comment:
A minor case: it is possible that an executionId conflicts with an execution name. For example: There is an execution with name `50`, and there is another execution in the same pipeline run has no name, but with executionId `50`. I guess it is a very minor situation, so feel free to take no action for now.

Reply:
I didn't think of this, that is a good point! If possible, it'd be nice to provide reasonable protection against this case in this PR, hopefully without changing the dropdown implementation.

For example, if there are any characters that the execution or artifact name does typically not allow, then I could update the text to include that character with the ID. Do execution or artifact names allow spaces? If not, I could change the above line to the following:
```
getExecutionName(executionArtifact.execution) ||
`ID: ${executionArtifact.execution.getId().toString()}`;
```
I would change the other corresponding references as well.